nixos/modules/services/networking/hylafaxp.nix

```diff
@@ -0,0 +1,187 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  setupScript = ''
+    ## locks
+    mkdir -p -m 0755 /var/lock/uucp
+    chown "uucp:uucp" "/var/lock/uucp"
+    ## etc
+    cp -r ${pkgs.hylafaxp}/share/hylafax/etc/hylafax /etc
+    chown -R "uucp:uucp" "/etc/hylafax"
+    chmod -R 0755 /etc/hylafax
+    ## spool
+    cp -r ${pkgs.hylafaxp}/share/hylafax/var/spool/hylafax /var/spool
+    cat ${etcConfig} - > /var/spool/hylafax/etc/config
+    cat ${etcHostsHfaxd} - > /var/spool/hylafax/etc/hosts.hfaxd
+    ${pkgs.ghostscript}/bin/gs -q -sDEVICE=tiffg3 -sFONTPATH="${pkgs.ghostscript}/share/ghostscript/fonts ${pkgs.ghostscript}/share/ghostscript/${pkgs.ghostscript.version}/Resource/Init ${pkgs.ghostscript}/share/ghostscript/${pkgs.ghostscript.version}/lib ${pkgs.ghostscript}/share/ghostscript/${pkgs.ghostscript.version}/Resource/Font ${pkgs.ghostscript}/share/ghostscript/fonts ${pkgs.ghostscript}/share/fonts/default/ghostscript ${pkgs.ghostscript}/share/fonts/default/Type1 ${pkgs.ghostscript}/share/fonts/default/TrueType" /var/spool/hylafax/bin/genfontmap.ps > /var/spool/hylafax/etc/Fontmap.HylaFAX 2>/dev/null
+    ${concatStringsSep ";" (zipListsWith (a: b: "cp " + a + " /var/spool/hylafax/etc/config." + b) etcConfigDevs etcDevs)};
+    chmod 0755 /var/spool/hylafax
+    ls -A /var/spool/hylafax/ | grep -v -E 'recvq|log' | ${pkgs.gawk}/bin/awk '{print "/var/spool/hylafax/"$1}' | xargs chmod -R 0755
```

Comment:
@Mic92 It's also good but I prefer it my way since it's easier to understand and maintain.

@joachifm I've considered this previously and decided I'd like to avoid bash-only features when possible.

If you guys can produce a solution that doesn't depend on regex matching (compute) or that walks only the one time (disk io), I'm sticking with the choice I find easiest to maintain.